### ISSUE TITLE
[Feature Request] Official CRUD action support for v-data-table

### ISSUE THREAD

**Issue by MatthewAry** [vuetifyjs/vuetify#2835]
### New Functionality
Right now there, is no official/recommended method for adding CRUD actions into v-data-table. Current implementations involve some sort of hack.

### Improvements
Current workarounds require some sort of hacking or misuse of the header array. Official support should resolve this.

### Bugs or Edge Cases it Helps Avoid
This is a very common UI pattern that has no official, documented support in Vuetify. This will help reduce support overhead and standardize the implementation.

**Comment by KaelWD** [comment]
Can you give an example of the api for this, and what the result would look like?

**Comment by MatthewAry** [comment]
For the API: The 'value' property in the objects in the array sent to the headers prop should allow a null value so it can, 1) not reference any data in the table, 2) not be sortable.

An example of the end result can be:
![screen shot 2017-12-20 at 11 47 43 am](https://user-images.githubusercontent.com/157217/34223252-da21bc02-e57b-11e7-88d9-231bf73cdedc.png)

Or can be like:
![screen shot 2017-12-20 at 11 50 28 am](https://user-images.githubusercontent.com/157217/34223286-0171be74-e57c-11e7-9ea8-a38f58b80b16.png)

With the data sent to the header prop looking something like this:
```
export default {
  // ...
  data () {
    return {
      // ...
      headers: [
        {
          text: 'ID',
          value: 'id',
          width: '80px'
        },
        {
          text: 'Customer Name',
          value: 'name',
          align: 'left'
        },
        {
          text: 'Actions',
          value: null
        }
      ],
      // ...
    }
  }
}
```

**Comment by ais-one** [comment]
Hi, this is a little more complicated than you might want.

The search needs to be more flexible, rather than just a single search text

You need to make the store configurable to handle crud from various data sources, or APIs

Also there is one part very complicated that I am dealing with right now which is handling relations, so other than standard widgets, you need to open another crud table with the ID as a filter for the child data.

**Comment by MatthewAry** [comment]
@ais-one 
- Columns with CRUD actions won't always need to be indexed for search. Thus, because it doesn't need indexing, it shouldn't be a problem.

- Search is already flexible. If you are talking about multiple search fields with filter controls, then you'll probably need to make a separate issue for it.

- On the store configurable part, I'm not sure what you mean on that. But it's probably something that needs to its own issue as well.

- On your last item... Did you know that the Vuetify project has a chatroom on Discord? There they have a need-help channel which you can get better support from.

**Comment by ais-one** [comment]
@MatthewAry 

I was just commenting with the aim of making this Issue usable in more real-life cases, if the decision is made to implement this issue.

- the search part I was mentioning about is something like listing records within a date range in the data-table (pagination required to handle large result set)

- Currently, I am using vuex store to maintain current page after navigating to another data-table, I also wanted to try and maintain the search filters when navigating away to another route but I have not been able to make this workable, and I do not want to go down the localStorage road on this.

a small part of what I am trying to achieve is this:

  routes: [
    {
      path: '/notes',
      name: 'notes',
      component: Notes,
      beforeEnter: AuthGuard,
      props: {
        parentKey: null,
        storeName: 'notes',
        _find, _findOne, _create, _update, _delete,
        defaultData, filterData,
        formatters, headers,
        FilterVue, FormVue
      },
      // Then another component here using a different Vuex module name

So basically, you can add custom things to handle your use case, e.g.

1) if _delete is null { _delete: null }, then there will be no delete button
2) FormVue lets you customize how your form should look like, same concept for FilterVue

I will go look at the Chatroom, but now still trying to clear the block I am facing on nested component...

<crud-component level="one" data="user">
  <crud-component level="two" data="post">
  </crud-component>
</crud-component>

Basically when you click to edit user, the form will open with editable fields for the user, and also a button where you can open another table to edit posts by that user, once done, when click back, will go back to that user edit form

**Comment by kevinqqnj** [comment]
Thanks @johnleider .
I tried the demo in https://next.vuetifyjs.com/components/data-tables#example-crud. It sounds good.
A minor bug:
When you click "Edit Action" -> Open the Edit Dialog -> Click "Close" or "Save", there's a flash of "New Item" appears in the title. 
Expected behavior: the title of Edit Dialog should not change to "New Item".

**Comment by ais-one** [comment]
Hi, I made a flexible CRUD based on data-tables, and it is able to do nested CRUD, filtering can have more fields and forms can be customised...

The repository is below:

[https://github.com/ais-one/vue-crud-x](https://github.com/ais-one/vue-crud-x)

More details are described in the article below:

https://medium.com/@aaronjxz/vue-crud-x-a-highly-customisable-crud-component-using-vuejs-and-vuetify-2b1539ce2054

The example backend is using firebase but it can be replaced by what you need it to do...
